File: libs/tests/helpers/generator/call.go
package generator

import (
	"time"

	"github.com/Netcracker/qubership-profiler-backend/libs/storage"

	"github.com/Netcracker/qubership-profiler-backend/libs/protocol/data"
)

func Convert(c data.Call) model.Call {
	call := model.Call{
		Time:              time.UnixMilli(c.Time),
		CpuTime:           c.CpuTime,
		WaitTime:          c.WaitTime,
		MemoryUsed:        c.MemoryUsed,
		Duration:          int32(c.Duration),
		NonBlocking:       c.NonBlocking,
		QueueWaitDuration: int32(c.QueueWaitDuration),
		SuspendDuration:   int32(c.SuspendDuration),
		Calls:             int32(c.Calls),
		Transactions:      int32(c.Transactions),
		LogsGenerated:     int64(c.LogsGenerated),
		LogsWritten:       int64(c.LogsWritten),
		FileRead:          int64(c.FileRead),
		FileWritten:       int64(c.FileWritten),
		NetRead:           int64(c.NetRead),
		NetWritten:        int64(c.NetWritten),
		Namespace:         c.Namespace,
		ServiceName:       c.ServiceName,
		PodName:           c.PodName,
		RestartTime:       time.UnixMilli(c.RestartTime),
		Method:            c.Method,
		Params:            c.Params,
		TraceFileIndex:    c.TraceFileIndex,
		BufferOffset:      c.BufferOffset,
		RecordIndex:       c.RecordIndex,
	}
	return call
}

File: apps/collector/src/main/java/com/netcracker/profiler/model/Call.java
package com.netcracker.profiler.model;

import java.util.List;
import java.util.Map;

public class Call {
    public long time;
    public long cpuTime;
    public long waitTime;
    public long memoryUsed;
    public int method;
    public int duration;
    public long nonBlocking;
    public int queueWaitDuration;
    public int suspendDuration;
    public int calls;
    public int traceFileIndex;
    public int bufferOffset;
    public int recordIndex;
    public long transactions;
    public int logsGenerated, logsWritten;
    public long fileRead, fileWritten;
    public long netRead, netWritten;
    public String threadName;
    public String callsStreamIndex;

    public Map<Integer, List<String>> params;

    public void setSuspendDuration(int suspendDuration) {
        this.suspendDuration = suspendDuration;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Call)) return false;

        Call call = (Call) o;

        if (bufferOffset != call.bufferOffset) return false;
        if (recordIndex != call.recordIndex) return false;
        if (traceFileIndex != call.traceFileIndex) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = traceFileIndex;
        result = 31 * result + bufferOffset;
        result = 31 * result + recordIndex;
        return result;
    }

    @Override
    public String toString() {
        return "Call{" +
                "time=" + time +
                ", cpuTime=" + cpuTime +
                ", waitTime=" + waitTime +
                ", memoryUsed=" + memoryUsed +
                ", method=" + method +
                ", duration=" + duration +
                ", queueWaitDuration=" + queueWaitDuration +
                ", suspendDuration=" + suspendDuration +
                ", calls=" + calls +
                ", traceFileIndex=" + traceFileIndex +
                ", bufferOffset=" + bufferOffset +
                ", recordIndex=" + recordIndex +
                ", transactions=" + transactions +
                ", logsGenerated=" + logsGenerated +
                ", logsWritten=" + logsWritten +
                ", fileRead=" + fileRead +
                ", fileWritten=" + fileWritten +
                ", netRead=" + netRead +
                ", netWritten=" + netWritten +
                ", threadName='" + threadName + '\'' +
                ", params=" + params +
                '}';
    }
}
